app/controllers/sign_up/select_email_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module SignUp
+  class SelectEmailController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+    before_action :verify_needs_completions_screen
+
+    def show
+      @sp_name = sp_name
+      @user_emails = user_emails
+      @last_sign_in_email_address = last_email
+      @select_email_form = build_select_email_form
+    end
+
+    def create
+      @select_email_form = build_select_email_form
+
+      result = @select_email_form.submit(form_params)
+      if result.success?
+        session[:selected_email_id] = form_params[:selected_email_id]
+        redirect_to sign_up_completed_path
+      else
+        flash[:error] = result.first_error_message
+        redirect_to sign_up_select_email_path
+      end
+    end
+
+    def sp_name
+      if current_sp
+        @sp_name ||= current_sp.friendly_name || sp.agency&.name
```

Comment:
Where is this `sp` method defined? Do we have test coverage for this branch of logic?

Reply:
On second thought, I'm wondering if I even need this function? Is a user ever going to see a select email to share with a partner and have no SP name set? That seems illogical.